plugins/woocommerce/src/Internal/ProductQueryFilters/Controller.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * MainQueryFilters class file.
+ */
+
+namespace Automattic\WooCommerce\Internal\ProductQueryFilters;
+
+use Automattic\WooCommerce\Internal\RegisterHooksInterface;
+use Automattic\WooCommerce\Internal\Traits\AccessiblePrivateMethods;
+use WC_Cache_Helper;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * MainQueryFilters class.
+ */
+class Controller implements RegisterHooksInterface {
+	use AccessiblePrivateMethods;
+
+	const TRANSIENT_GROUP = 'filter_data';
```

Comment:
Can you add this (and any other transient being used for product cache) to `wc_delete_product_transients` function as well? it will make it easier for folks to refresh data.

Reply:
@vedanshujain Great point! I address this in https://github.com/woocommerce/woocommerce/pull/43772/commits/7c7f414702d52a12220c91780a572bd2361ba881. All transients set by the new service provider use `filter_data` version, so I think hooking into `woocommerce_delete_product_transients` action is enough. Let me know if there are other parts else I need to cover.